File: Suyeon/ch14_Tree/226_Invert_Binary_Tree.py
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:

        def dfs(root):
            if root is None:
                return
            dfs(root.left)
            dfs(root.right)
            root.left, root.right = root.right, root.left

        dfs(root)

        return root


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root:
            root.left , root.right = self.invertTree(root.right), self.invertTree(root.left)

            return root

File: Sujin/Ch13_Graph&Tree&Dijkstra/226_Invert_Binary_Tree.cpp
#include <iostream>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    void DFS(TreeNode* root){
        if(root==NULL) return;

        DFS(root->left);
        DFS(root->right);

        TreeNode* tmp = root->left;
        root->left = root->right;
        root->right = tmp;
    }

    TreeNode* invertTree(TreeNode* root) {
        if(root==NULL) return NULL; // 예외처리

        TreeNode* res = root;
        DFS(root);
        return res;
    }
};